File: Binary Search Algorithm/JavaScript/Binary-Search-Algorithm.js
function binarySearch(arr, x) {
    let left = 0;
    let right = arr.length - 1;

    while (left <= right) {
        const mid = left + Math.floor((right - left) / 2);

        if (arr[mid] === x) {
            return mid;
        } else if (arr[mid] < x) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    return -1;
}

const arr = [2, 3, 4, 10, 40];
const x = 10;

const result = binarySearch(arr, x);
if (result !== -1) {
    console.log(`Element found at index ${result}`);
} else {
    console.log("Element not found in the array");
}

File: website/algorithms/binary-search-algorithm/Binary_Search_Algorithm.rs
fn binary_search(arr: &[i32], x: i32) -> i32 {
    let mut left = 0;
    let mut right = arr.len() as i32 - 1;

    while left <= right {
        let mid = left + (right - left) / 2;

        if arr[mid as usize] == x {
            return mid;
        } else if arr[mid as usize] < x {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    -1
}

fn main() {
    let arr = vec![2, 3, 4, 10, 40];
    let x = 10;

    let result = binary_search(&arr, x);
    if result != -1 {
        println!("Element found at index {}", result);
    } else {
        println!("Element not found in the array");
    }
}
